python/src/vmaf/core/feature_assembler.py

```diff
@@ -87,18 +88,18 @@ def run(self):
 
         # assemble an output dict with demanded atom features
         # atom_features_dict = self.fextractor_atom_features_dict
-        result_dicts = map(lambda x: dict(), self.assets)
+        result_dicts = to_list(map(lambda x: dict(), self.assets))
         for fextractor_type in self.feature_dict:
             assert fextractor_type in self.type2results_dict
             for atom_feature in self._get_atom_features(fextractor_type):
                 scores_key = self._get_scores_key(fextractor_type, atom_feature)
                 for result_index, result in enumerate(self.type2results_dict[fextractor_type]):
                     result_dicts[result_index][scores_key] = result[scores_key]
 
-        self.results = map(
-            lambda (asset, result_dict): BasicResult(asset, result_dict),
+        self.results = to_list(map(
+            lambda tasset: BasicResult(tasset[0], tasset[1]),
```

Comment:
`lambda (a, b): ...` is a syntax error in python3.
See for example http://cheng.logdown.com/posts/2016/11/30/python3-lambda-parameter